jose.py

```diff
@@ -305,27 +315,32 @@ def b64decode_url(istr):
     """ JWT Tokens may be truncated without the usual trailing padding '='
         symbols. Compensate by padding to the nearest 4 bytes.
     """
-    istr = encode_safe(istr)
     try:
-        return urlsafe_b64decode(istr + '=' * (4 - (len(istr) % 4)))
-    except TypeError as e:
+        return urlsafe_b64decode(istr + six.b('=') * (4 - (len(istr) % 4)))
+    except (TypeError, binascii.Error) as e:
         raise Error('Unable to decode base64: %s' % (e))
 
 
 def b64encode_url(istr):
     """ JWT Tokens may be truncated without the usual trailing padding '='
         symbols. Compensate by padding to the nearest 4 bytes.
     """
-    return urlsafe_b64encode(encode_safe(istr)).rstrip('=')
+    return urlsafe_b64encode(encode_safe(istr)).rstrip(six.b('='))
 
 
-def encode_safe(istr, encoding='utf8'):
-    try:
-        return istr.encode(encoding)
-    except UnicodeDecodeError:
-        # this will fail if istr is already encoded
-        pass
-    return istr
+if six.PY3:
```

Comment:
we might want to refactor this style: http://astrofrog.github.io/blog/2016/01/12/stop-writing-python-4-incompatible-code/